include/seqan3/io/sequence_file/format_genbank.hpp

```diff
@@ -0,0 +1,252 @@
+ * \implements  seqan3::SequenceFileOutputFormat
+ * \ingroup     sequence
+ *
+ * \details
+ *
+ * ### Introduction
+ *
+ * genbank is the format used in GenBank sequence database. See
+ * (http://quma.cdb.riken.jp/help/gbHelp.html) for a an in-depth description of the format.
+ *
+ * ### Fields
+ *
+ * The genbank format provides the fields seqan3::field::SEQ and seqan3::field::ID. Both fields are required when
+ * writing.
+ *
+ * ### Implementation notes
+ *
+ * There is no truncate_ids option while reading because the GenBank format has no (FASTA-like) idbuffer. Instead,
+ * there is the option "complete_header" to indicate whether the whole header is to be read
+ * (embl_genbank_complete_header=true) or only the "LOCUS" information should be stored.
+ *
+ * Passed qualities to either the read or write function are ignored.
+ *
+ */
+class sequence_file_format_genbank
+{
+public:
+    /*!\name Constructors, destructor and assignment
+     * \{
+     */
+     //!\brief Default constructor is explicitly defaulted, you need to provide a stream or file name.
+    sequence_file_format_genbank() = default;
+    //!\brief Copy construction is explicitly deleted, because you can't have multiple access to the genbank file.
+    sequence_file_format_genbank(sequence_file_format_genbank const &) = delete;
+    //!\brief Copy assignment is explicitly deleted, because you can't have multiple access to the genbank file.
+    sequence_file_format_genbank & operator=(sequence_file_format_genbank const &) = delete;
+    //!\brief Move construction is defaulted.
+    sequence_file_format_genbank(sequence_file_format_genbank &&) = default;
+    //!\brief Move assignment is defaulted.
+    sequence_file_format_genbank & operator=(sequence_file_format_genbank &&) = default;
+    //!\}
+
+    //!\brief The valid file extensions for this format; note that you can modify this value.
+    static inline std::vector<std::string> file_extensions
+    {
+        { "genbank" },
+        { "gb" },
+        { "gbk" },
+    };
+
+    //!\copydoc SequenceFileInputFormat::read
+    template <typename stream_type,     // constraints checked by file
+              typename seq_legal_alph_type, bool seq_qual_combined,
+              typename seq_type,        // other constraints checked inside function
+              typename id_type,
+              typename qual_type>
+    void read(stream_type                                                            	& stream,
+              sequence_file_input_options<seq_legal_alph_type, seq_qual_combined> const & options,
+              seq_type                                                                  & sequence,
+              id_type                                                                   & id,
+              qual_type                                                                 & SEQAN3_DOXYGEN_ONLY(qualities))
+    {
+        using stream_char_t = typename stream_type::char_type;
```

Comment:
```suggestion
        using stream_buff_t = std::istreambuf_iterator<typename stream_type::char_type>;
```
and then replace the corresponging partes in the stream_view definition. makes it much shorter :)